pkg/lib/packagekit.js

```diff
@@ -186,30 +187,21 @@ export function watchTransaction(transactionPath, signalHandlers, notifyHandler)
  *        .catch(ex => { handle exception });
  */
 export function transaction(method, arglist, signalHandlers, notifyHandler) {
-    return new Promise((resolve, reject) => {
-        call("/org/freedesktop/PackageKit", "org.freedesktop.PackageKit", "CreateTransaction", [])
-                .done(result => {
-                    const transactionPath = result[0];
-                    let watchPromise;
-                    if (signalHandlers || notifyHandler)
-                        watchPromise = watchTransaction(transactionPath, signalHandlers, notifyHandler);
-                    if (!watchPromise)
-                        watchPromise = cockpit.resolve();
-
-                    watchPromise
-                            .done(() => {
-                                if (method) {
-                                    call(transactionPath, transactionInterface, method, arglist)
-                                            .done(() => resolve(transactionPath))
-                                            .fail(reject);
-                                } else {
-                                    resolve(transactionPath);
-                                }
-                            })
-                            .fail(reject);
-                })
-                .fail(reject);
-    });
+    return call("/org/freedesktop/PackageKit", "org.freedesktop.PackageKit", "CreateTransaction", [])
+            .then(([transactionPath]) => {
+                if (!signalHandlers && !notifyHandler)
+                    return transactionPath;
+
+                const watchPromise = watchTransaction(transactionPath, signalHandlers, notifyHandler) || Promise.resolve();
+                watchPromise.then(() => {
+                    if (method) {
+                        return call(transactionPath, transactionInterface, method, arglist)
+                                .then(() => transactionPath);
+                    } else {
+                        return transactionPath;
+                    }
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18155-20230112-090223-297b7da2-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)